app/services/atmosphere/billing_service.rb

```diff
@@ -20,45 +20,57 @@ def self.bill_all_appliances
+        # Figure out how long this deployment can continue to run without being billed again.
+        billing_interval = (billing_time - dep.prepaid_until) # This will return time in seconds
         if billing_interval < 0
           # The appliance is still prepaid - nothing to be done.
-          Rails.logger.debug("Appliance #{appl.id} is prepaid until #{appl.prepaid_until}. Nothing to be done.")
+          Rails.logger.debug("Deployment #{dep.id} belonging to appliance #{appl.id} is prepaid until #{dep.prepaid_until}. Nothing to be done.")
         else
-          # Bill the hell out of this appliance! :)
-          Rails.logger.debug("Billing appliance #{appl.id} for #{(billing_interval/3600)} hours of use.")
+          # Bill the hell out of this deployment! :)
+          Rails.logger.debug("Billing deployment #{dep.id} belonging to appliance #{appl.id} for #{(billing_interval/3600)} hours of use.")
           # Open a transaction to avoid race conditions etc.
           ActiveRecord::Base.transaction do
-            amount_due = calculate_charge_for_appliance(appl, billing_time, apply_prepayment)
+            amount_due = calculate_charge_for_deployment(dep, billing_time, apply_prepayment)
             # Check if there are sufficient funds
             if (amount_due > appl.fund.balance-appl.fund.overdraft_limit)
-              # We've run out of funds. This appliance cannot be paid for. Flagging it as expired.
-              Rails.logger.warn("The balance of fund #{appl.fund.id} is insufficient to cover continued operation of appliance #{appl.id} (current balance: #{appl.fund.balance}; overdraft limit: #{appl.fund.overdraft_limit}; calculated charge: #{amount_due}). Flagging appliance as expired.")
-              BillingLog.new(timestamp: Time.now.utc, user: appl.appliance_set.user, appliance: appl.id.to_s+'---'+(appl.name.blank? ? 'unnamed_appliance' : appl.name), fund: appl.fund.name, message: "Funding expired for this appliance.", actor: "bill_appliance", amount_billed: 0).save
-              appl.billing_state = "expired" # A separate method will be used to clean up all expired appliances according to their funds' termination policies
-              appl.save
+              # We've run out of funds. This deployment cannot be paid for. Flagging it as expired.
+              Rails.logger.warn("The balance of fund #{appl.fund.id} is insufficient to cover continued operation of deployment #{dep.id} belonging to appliance #{appl.id} (current balance: #{appl.fund.balance}; overdraft limit: #{appl.fund.overdraft_limit}; calculated charge: #{amount_due}). Flagging deployment as expired.")
+              BillingLog.new(timestamp: Time.now.utc, user: appl.appliance_set.user, appliance: appl.id.to_s+'---'+(appl.name.blank? ? 'unnamed_appliance' : appl.name), fund: appl.fund.name, message: "Funding expired for deployment #{dep.id}.", actor: "bill_appliance", amount_billed: 0).save
```

Comment:
Line is too long. [292/80]<br>Surrounding space missing for operator '+'.<br>Prefer single-quoted strings when you don't need string interpolation or special symbols.